### ISSUE TITLE
VimwikiRenameLink not working with Markdown

### ISSUE THREAD

**Issue by davidlmontgomery** [vimwiki/vimwiki#237]
I find that VimwikiRenameLink doesn't work when using Markdown links like this 

`[Link text](link-file)
`

Because I am only using this kind of Markdown link (I am not using reference-style links), for my purposes the following hack worked to fix this issue. But it isn't a general solution.

```
--- vimwiki_markdown_custom.vim 2016-07-05 20:17:09.084795600 -0700
+++ vimwiki_markdown_custom.vim 2016-07-05 20:16:22.290611000 -0700
@@ -31,8 +31,8 @@
 " 1. [URL][], or [DESCRIPTION][URL]

 let s:wikilink_md_prefix = '['
-let s:wikilink_md_suffix = ']'
-let s:wikilink_md_separator = ']['
+let s:wikilink_md_suffix = ')'
+let s:wikilink_md_separator = ']('
 let s:rx_wikilink_md_prefix = vimwiki#u#escape(s:wikilink_md_prefix)
 let s:rx_wikilink_md_suffix = vimwiki#u#escape(s:wikilink_md_suffix)
 let s:rx_wikilink_md_separator = vimwiki#u#escape(s:wikilink_md_separator)

```

**Comment by EinfachToll** [comment]
Depends on #256

**Comment by ranebrown** [comment]
Renaming a page should now work but the links are not correctly updated. Related to #406

**Comment by ranebrown** [comment]
@davidlmontgomery This should be fixed on the latest `dev` branch.

**Comment by f-dj** [comment]
> @davidlmontgomery This should be fixed on the latest `dev` branch.
Can confirm that it works in most cases, but still does not work with whitespaces. Is this expected behaviour?

I mean, I was surprised that whitespaces in links work in the first place. But shouldn't the renaming work also for consistency if you are allowed to generate them automatically (by going to visual mode, select multiple words, hit enter)? Feels like a minor bug.

**Comment by f-dj** [comment]
I opened PR #824 to make the `find_files` function work for filenames with whitespaces. However, on master the `split` method already took `'\n'` as  2nd parameter. So it should only affect the dev branch. 

Would be good to merge this PR anyway IMO, so that the dev branch is as functional as master. Also, this prevents a bug from being introduced if dev is eventually merged into master.

**Comment by etiennekintzler** [comment]
The issue still appear in my case (file is renamed but links are not updated). 

This is the case _whether there is a space or not_ in the filename. 

The lines in my `.vimrc` related to Vimwiki:

```vim
" vim wiki path
let wiki_life = {'path' : 'path/to/folder', 'syntax' : 'markdown', 'ext': '.md'}
let g:vimwiki_list = [wiki_life]
```

**Comment by tinmarino** [comment]
@etiennekintzler  Use the dev branch

**Comment by etiennekintzler** [comment]
@tinmarino thanks ! I thought it has been incorporated in the last release.

I updated to the `dev` branch using Vim-Plug and I have now the following error when opening my wiki :

```vim
Error detected while processing function vimwiki#vars#get_global:
line    1:
E716: Key not present in Dictionary: schemes_local
E15: Invalid expression: g:vimwiki_global_vars[a:key]
Press ENTER or type command to continue
```
